unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,156 @@
+import datetime as dt
+import logging
+from dataclasses import dataclass
+from typing import Any, Generator, Optional
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig]
+    channels: Optional[list[str]] = Field(
+        default=None, description="List of Discord channel IDs to process"
+    )
+
+
+class DiscordIndexerConfig(IndexerConfig):
+    pass
+
+
+@dataclass
+class DiscordIndexer(Indexer):
+    connection_config: DiscordConnectionConfig
+    index_config: DiscordIndexerConfig
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        client = self.get_client()
+        channels_to_process: set[str] = set(self.connection_config.channels or [])
+
+        for channel_id in list(channels_to_process):
+            file_data = self.get_channel_file_data(channel_id=channel_id, client=client)
+            if file_data:
+                yield file_data
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_channel_file_data(self, channel_id: str, client) -> Optional[FileData]:
+        # Fetch channel metadata
+        dt.datetime.utcnow().isoformat()
+        identifier = channel_id
+        channel_id = f"{channel_id}.txt"
+        source_identifiers = SourceIdentifiers(
+            filename=identifier,
```

Comment:
This filename won't contain the file extension.

Reply:
@rbiseck3 That was on purpose, when doing filename = channel_id, I was having issues

Reply:
What issues were you having? This is the typical pattern we use elsewhere and might break how the uploader writes the file.

Reply:
The structured-output gets written as `.txt.json` when doing the changes you requested.
![image](https://github.com/user-attachments/assets/18cfa7eb-2bf7-404e-919c-ad1aec92d7f0)